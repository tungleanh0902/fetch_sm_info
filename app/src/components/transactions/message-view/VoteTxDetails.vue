<template>
  <div class="tx__content">
    <TransactionIcon :transaction-type="type" />
    <div class="tx__content__left">
      <h3>{{ caption }}</h3>
      <span
        ><i class="italic">{{ transaction.details.voteOption }}</i>
        &nbsp;on&nbsp;</span
      >
      <router-link
        :to="{
          name: 'proposal',
          params: { proposalId: String(transaction.details.proposalId) },
        }"
        >Proposal &#35;{{ transaction.details.proposalId }}</router-link
      >
    </div>
  </div>
</template>

<script>
import TransactionIcon from "../TransactionIcon"

export default {
  name: `vote-tx-details`,
  components: {
    TransactionIcon,
  },
  props: {
    transaction: {
      type: Object,
      required: true,
    },
    validators: {
      type: Object,
      required: true,
    },
  },
  data: () => {
    return {
      type: `Voted`,
      caption: `Voted`,
    }
  },
}
</script>
<style scoped>
.italic {
  font-style: italic;
}
</style>
