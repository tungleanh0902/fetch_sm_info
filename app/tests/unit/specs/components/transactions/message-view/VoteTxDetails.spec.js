import { shallowMount } from "@vue/test-utils"
import VoteTxDetails from "src/components/transactions/message-view/VoteTxDetails"

describe(`VoteTxDetails`, () => {
  let wrapper

  const tx = {
    type: "VoteTx",
    hash: "A0DEB29E97A4DF38289D55D63C5724588985E1D35B26518CB66EAF96CFEF2E04",
    height: 779965,
    timestamp: "2020-02-10T10:15:51Z",
    memo: "",
    success: true,
    fees: [],
    details: {
      proposalId: 17,
      voteOption: "Yes",
    },
  }

  it(`renders a vote transaction message`, () => {
    wrapper = shallowMount(VoteTxDetails, {
      propsData: {
        transaction: tx,
        validators: {},
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
