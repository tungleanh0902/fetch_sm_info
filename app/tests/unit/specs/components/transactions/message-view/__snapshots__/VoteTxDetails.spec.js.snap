// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteTxDetails renders a vote transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Voted"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Voted
    </h3>
     
    <span>
      <i
        class="italic"
      >
        Yes
      </i>
      
       on 
    </span>
     
    <router-link-stub
      to="[object Object]"
    >
      Proposal #17
    </router-link-stub>
  </div>
</div>
`;
